### PR TITLE
feat: Add emitEvent step + cleanup

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Generic step to emit event in a workflow, allow no input in order to be able to by pass it if a constraint would like to prevent emitting an event based on some contextual data
- Cleanup all `service` from messages to be `source` which makes more sense and is more align with general event formatting and more agnostic

**note**
This does not include the event group id work to validate it is properly present in the transaction metadata and the place it gets either generated or captured

FIXES CORE-2266

### CODE REVIEW

File: packages/core/core-flows/src/common/steps/emit-event.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { composeMessage, ModuleRegistrationName } from "@medusajs/utils"
+import { createStep, StepExecutionContext } from "@medusajs/workflows-sdk"
+
+type Input = {
+  eventName: string
+  source: string
+  object: string
+  action?: string
+  options?: Record<string, any>
```
Comment (riqwan) [inline packages/core/core-flows/src/common/steps/emit-event.ts:9]: what are these here?

File: packages/core/core-flows/src/common/steps/release-event.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ModuleRegistrationName } from "@medusajs/utils"
+import { createStep } from "@medusajs/workflows-sdk"
+
+export const releaseEventsStepId = "emit-event-step"
```
Comment (sradevski) [inline packages/core/core-flows/src/common/steps/release-event.ts]:
```suggestion
export const releaseEventsStepId = "release-event-step"
```

File: packages/core/core-flows/src/common/steps/emit-event.ts (hunk 1 above)
Comment (adrien2p) [reply packages/core/core-flows/src/common/steps/emit-event.ts:9]: potential options for the choosen event bus

File: packages/core/core-flows/src/common/steps/release-event.ts (hunk 2 above)
Comment (adrien2p) [reply packages/core/core-flows/src/common/steps/release-event.ts]: good catch 😂

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**